operator/controllers/csiscaleoperator_controller.go

```diff
@@ -1553,3 +1562,55 @@ func (r *CSIScaleOperatorReconciler) resourceExists(instance *csiscaleoperator.C
 		return true, nil
 	}
 }
+
+// getConfigMap fetches data from the "ibm-spectrum-scale-csi-config" configmap from the cluster
+// and returns a configmap reference.
+func (r *CSIScaleOperatorReconciler) getConfigMap(instance *csiscaleoperator.CSIScaleOperator, name string) (*corev1.ConfigMap, error) {
+
+	logger := csiLog.WithName("getCSIConfig").WithValues("Kind", corev1.ResourceConfigMaps, "Name", name)
+	logger.Info("Read CSI configmap from the cluster.")
+
+	cm := &corev1.ConfigMap{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      name,
+		Namespace: instance.Namespace,
+	}, cm)
+
+	if err != nil && errors.IsNotFound(err) {
+		message := "Resource not found."
```

Comment:
An info message can be improved here like: "An optional configmap <config map name> not found" so that it helps in debug